text/0000-network-transform-buffered-interpolation.md

```diff
@@ -0,0 +1,408 @@
+After this, users should be able to select the interpolation algorithm that best fits the GameObject they want to sync.
+
+In addition, this RFC suggests removing the 
+```C#
+        public AnimationCurve DistanceSendrate = AnimationCurve.Constant(0, 500, 20);
+```
+configuration (that was used to set interpolation times) to use instead a swappable interpolator.
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+NetworkTransform now has this new configuration:
+- `Interpolator`: Allows users to select the interpolation algorithm they want to use. This interpolator will interpolate "ghost" side state changes. Note it'll still be up to users to interpolate their authoritative objects however they want. If for example Client 1 is authoritative over a player object, this RFC will add interpolation to the server object and the other clients' objects, not Client 1's object. It'll still be up to users to interpolate that authoritative object (using RigidBody's interpolation for example or anything else).
+
+
+A default `BufferedLinearInterpolator` is provided. The buffered linear interpolator will buffer values before making them available to NetworkTransform's value update. This will allow NetworkTransform to accumulate jittered network values without affecting the transform's state and latter be able to consume them at regular intervals.
+
+This will smooth tranforms on jittery connections, however it'll add more latency to your transform updates. This should be used carefully as to not add more latency than needed to your game. Bigger buffer values could be expected when targeting mobile platforms for example, while keeping interpolation times lower for platforms with typically more stable connections.
+
+You can find the interpolator's configuration in its associated scriptable object factory.
+It is advised to use the same interpolator for all your NetworkTransforms. This way, you're making sure all your objects will be synchronized and use the same delays. Having an interpolator with 100ms delay and another with 500ms delay could cause visible overlaps and desyncs where an object at time say `t=10s` tries to interact with an object buffered at time `t=9.6s`.
+
+If you have a case where you'd like to use a different configuration, you can create a new instance of the interpolator scriptable object and assign it to your object. If you want a new default value for all your NetworkTransforms, you can assign a scriptable object as default value for a class in the Unity Editor, by selecting the C# script and assigning the default value in the script's inspector (this will add a reference in the C# script's meta file).
+
+If you don't want interpolation, you can use the `NoInterpolation` interpolator which will take in new values and present them directly when asked, without doing anything else.
```

Comment:
I assume leaving the factory serialize field empty on NetworkTransform will default to this `NoInterpolation`?

Reply:
I would do smooth by default, so I'd use BufferedInterpolator by default.
I'll check with UX for a final approval, but on your side, what do you think?

Reply:
FWIW, I think leaving interpolator field empty/null should not default to `BufferedInterpolator` but instead default to latest pos/rot/sca snapping.

Reply:
I'm curious, why?
I'm not strongly in favor of any one, I wanna build a list of pros and cons I can expose to UX.
For me:
Pros:
it's smooth by default
Cons:
It adds latency by default

Reply:
I actually want to move this conversation from here to completely different space :) I'll comment over there at the motivations section soon :P